applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -1029,8 +1029,8 @@ static void new_config_handle(struct cloud_data_cfg *new_config)
 }
 
 /**
- * @brief Function that requests A-GPS and P-GPS data upon receiving a request from the GNSS module.
- *	  If both A-GPS and P-GPS is enabled. A-GPS will take precedence.
+ * @brief Function that requests A-GPS and P-GPS data upon receiving a request from the
+ *        location module. If both A-GPS and P-GPS are enabled, A-GPS will take precedence.
```

Comment:
When both A-GPS and P-GPS are enabled,  parts of each are used.

The A-GPS assistance records except for almanac and ephemerides, and ephemerides from P-GPS are used together.  If A-GPS assistance cannot be retrieved due to connectivity issues, then time and location assistance records are estimated by the P-GPS library and used instead.

Reply:
Good question if they can be used together with the new integration.

Reply:
> Can A-GPS and P-GPS be enabled at the same time in the new integration?
> The futures were at some point mutually exclusive in the application

I'm not sure if I've changed this and compilation with pgps overlay leaves also `CONFIG_NRF_CLOUD_AGPS` enabled.
So in this function, `nrf_cloud_pgps_notify_prediction()` would be called if somebody would compile with a configuration (AGPS=n, PGPS=y) not used in `sample.yaml`.

> In case, `If both A-GPS and P-GPS are enabled, A-GPS will take precedence`. is not true.

I think this is true based on the implementation of this function. However, I think there might be other triggers for PGPS from location library.

Reply:
I need to make trials if this code block in this function actually makes a difference now that location library sends separate events for AGPS (LOCATION_EVT_GNSS_ASSISTANCE_REQUEST) and PGPS (LOCATION_EVT_GNSS_PREDICTION_REQUEST).

```
#if defined(CONFIG_NRF_CLOUD_PGPS)
	/* A-GPS data is not expected to be received. Proceed to schedule a callback when
	 * P-GPS data for current time is available.
	 */
	err = nrf_cloud_pgps_notify_prediction();
	if (err) {
		LOG_ERR("Requesting notification of prediction availability, error: %d", err);
	}
#endif
```

@tokangas, @hiltunent Do you think this block is needed?

Reply:
There is some misalignment now between how A-GPS and P-GPS requests are handled:

![Untitled (1)](https://user-images.githubusercontent.com/42843330/201857594-ab845362-eaea-4cb4-bfa8-87b052a84098.png)

Ideally they should take the same path. Maybe the best approach is to clean this up in a follow up PR.

Reply:
Yep, I'll create a separate ticket on this.